pokemon-spring-mvc/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/mvc/controllers/TrainerController.java

```diff
@@ -0,0 +1,211 @@
+    ResourceBundle messages = ResourceBundle.getBundle("Texts", LocaleContextHolder.getLocale());
+
+    @Inject
+    TrainerFacade trainerFacade;
+
+    @Inject
+    PokemonFacade pokemonFacade;
+
+    /**
+     * Controller method for listing all trainers
+     * @param model data to be displayed
+     * @return JSP page
+     */
+    @RequestMapping(value = "/list", method = RequestMethod.GET)
+    public String list(Model model) {
+        log.debug("mvc list()");
+        model.addAttribute("allTrainers", trainerFacade.getAllTrainers());
+        return "trainer/list";
+    }
+
+    /**
+     * Controller method for detailed trainer
+     * @param id of the trainer
+     * @param model data to be displayed
+     * @return JSP page name
+     */
+    @RequestMapping(value = "/detail/{id}", method = RequestMethod.GET)
+    public String detail(@PathVariable long id,
+                         Model model,
+                         RedirectAttributes redirectAttributes,
+                         UriComponentsBuilder uriComponentsBuilder) {
+        log.debug("mvc detail({})", id);
+
+        if (trainerFacade.getTrainerWithId(id) == null) {
+            redirectAttributes.addFlashAttribute("alert_danger", String.format(messages.getString("trainer.does.not.exists"), id));
+            return "redirect:" + uriComponentsBuilder.path("/trainer/list").build().encode().toUriString();
+        }
+
+        model.addAttribute("trainer", trainerFacade.getTrainerWithId(id));
+        try {
+            model.addAttribute("pokemons", getPokemons(pokemonFacade.getPokemonOfTrainer(id)));
+        } catch (NoSuchEntityException ex) {
+            redirectAttributes.addFlashAttribute("alert_warning", String.format(messages.getString("entity.does.not.exists"), "Trainer", id));
+            return "redirect:" + uriComponentsBuilder.path("/trainer/list").build().encode().toUriString();
+        }
+        return "trainer/detail";
+    }
+
+    /**
+     * GET controller for renaming of trainer
+     * @param id of trainer to be renamed
+     * @param model data to be displayed
+     * @return JSP page
+     */
+    @RequestMapping(value = "/rename/{id}", method = RequestMethod.GET)
+    public String rename(@PathVariable long id,
+                         Model model,
+                         RedirectAttributes redirectAttributes,
+                         UriComponentsBuilder uriComponentsBuilder) {
+        log.debug("mvc GET rename({})", id);
+
+        if (trainerFacade.getTrainerWithId(id) == null) {
+            redirectAttributes.addFlashAttribute("alert_danger", String.format(messages.getString("trainer.does.not.exists"), id));
+            return "redirect:" + uriComponentsBuilder.path("/trainer/list").build().encode().toUriString();
+        }
+
+        TrainerRenameDTO trainerRenameDTO = new TrainerRenameDTO();
+        trainerRenameDTO.setTrainerId(id);
+        model.addAttribute("trainerToUpdate", trainerRenameDTO);
```

Comment:
If the field (and getter and setter) names match between TrainerDTO and TrainerRenameDTO, you could avoid having to initialize TrainerRenameDTO and you could just pass in a fetched TrainerDTO. This **might** (I haven't tested) also have the pleasant side effect of prefilling name and surname to current values.